01.md

```diff
@@ -42,136 +47,161 @@ To obtain the `event.id`, we `sha256` the serialized event. The serialization is
 
-The `since` and `until` properties can be used to specify the time range of events returned in the subscription. If a filter includes the `since` property, events with `created_at` greater than or equal to `since` are considered to match the filter. The `until` property is similar except that `created_at` must be less than or equal to `until`. In short, an event matches a filter if `since <= created_at <= until` holds.
+The `ids`, `authors`, `#e`, and `#p` filter lists MUST contain exact 64-character lowercase hex values.
 
-All conditions of a filter that are specified must match for an event for it to pass the filter, i.e., multiple conditions are interpreted as `&&` conditions.
+The `since` and `until` properties are used to specify the time range of events returned in the subscription. An event matches the filter if `since <= created_at <= until` holds.
 
-A `REQ` message may contain multiple filters. In this case, events that match any of the filters are to be returned, i.e., multiple filters are to be interpreted as `||` conditions.
+The `limit` property informs the maximum number of events to return, sorted by created_at desc. It operates over the previously-stored events and is ignored afterward. 
 
-The `limit` property of a filter is only valid for the initial query and MUST be ignored afterwards. When `limit: n` is present it is assumed that the events returned in the initial query will be the last `n` events ordered by the `created_at`. It is safe to return less events than `limit` specifies, but it is expected that relays do not return (much) more events than requested so clients don't get unnecessarily overwhelmed by data.
+### Receiving Events
 
-### From relay to client: sending events and notices
+Relays send 3 types of messages during a subscription:  
 
-Relays can send 4 types of messages, which must also be JSON arrays, according to the following patterns:
-
-  * `["EVENT", <subscription_id>, <event JSON as defined above>]`, used to send events requested by clients.
-  * `["OK", <event_id>, <true|false>, <message>]`, used to indicate acceptance or denial of an `EVENT` message.
+  * `["EVENT", <subscription_id>, <JSON-serialized string of the event>]`, used to send events requested by clients.
   * `["EOSE", <subscription_id>]`, used to indicate the _end of stored events_ and the beginning of events newly received in real-time.
-  * `["CLOSED", <subscription_id>, <message>]`, used to indicate that a subscription was ended on the server side.
+  * `["CLOSED", <subscription_id>, <prefix>:<message>]`, used to indicate that a subscription was ended on the server side.
```

Comment:
why isn't the `OK` message listed here?

Reply:
I split them into a subscription protocol and a publishing protocol.